### PR TITLE
Discover and Movies Index Page

### LINKED ISSUE #7: Discover Page
As an authenticated user,
When I visit the '/discover' path
I should see
- [x] Button to Discover top 40 movies

**Details** When the user clicks on the top 40 button they should be taken to the movies page.

- [x] A text field to enter keyword(s) to search by movie title
- [x] A Button to Search by Movie Title

**Details** When the user clicks on the Search button they should be taken to the movies page

The movies will be retrieved by consuming [The MovieDB API](https://developers.themoviedb.org/3/getting-started/introduction)

### LINKED ISSUE #9: Movies Page
As an authenticated user,
When I visit the movies page,
I should see the 40 results from my search,
I should also see the "Find Top Rated Movies" button and the Find Movies form at the top of the page.

**Details:** The results from the search should appear on this page, and there should only be a maximum of 40 results. The following details should be listed for each movie.
- [x] Title (As a Link to the Movie Details page)
- [x] Vote Average of the movie

### PR DESCRIPTION
**Pull request by smcunning**
- Adds VCR and Webmock to Gemfile 
- Adds specs for discover and movies index pages 
- Adds discover routes, controller and view page 
- Adds movies routes, controller and view page 
- Updates discover index spec with VCR cassettes
- All tests passing

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -79,6 +80,8 @@ GEM
     faraday_middleware (1.0.0)
       faraday (~> 1.0)
     ffi (1.13.1)
+    figaro (1.2.0)
+      thor (>= 0.14.0, < 2)
```
Comment (KieraAllen) [inline Gemfile.lock:84]: By the power of Mjölnir!

File: app/controllers/movies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class MoviesController < ApplicationController
+  def index
+    if params[:search] == ""
+      flash[:error] = "Must provide a search query"
+      redirect_to "/discover"
+    elsif params[:search]
+      # search_term = params[:search]
+      # require "pry"; binding.pry
```
Comment (KieraAllen) [inline app/controllers/movies_controller.rb]: We'll just need to remove these pry comments (lines 8, 12, and 19), and maybe the one on line 7 (I can't remember if we were leaving that for our future selves).

File: app/views/movies/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+<%= form_tag "/movies", method: :get %>
+  <%= search_field_tag(:search, params[:search]) %>
+<%= submit_tag ("Search By Title") %>
+
+
+  <% @movies.each do |movie| %>
+    <div class="movie">
+      <div class="title">
+       <%= link_to "#{movie[:title]}" %>
```
Comment (KieraAllen) [inline app/views/movies/index.html.erb]: We might not need to string interpolate here, but the path we add (for that movie's show page) will have quotes. For example:
`<%= link_to item.name, "/items/#{item.id}" %>`

File: spec/features/discover_spec.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+describe 'as a user on the discover page' do
```
Comment (KieraAllen) [inline spec/features/discover_spec.rb]: SUPER minor, but maybe we change this describe to 'as a REGISTERED user...' just so it's clear

File: spec/features/movies/index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+describe 'as a user on the movies index page' do
```
Comment (KieraAllen) [inline spec/features/movies/index_spec.rb]: Same change as in the discover_spec: add "registered" before user

File: spec/features/movies/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      within(first(".movie")) do
+        name = find('.title').text
+        expect(name).not_to be_empty
+      end
+    end
+  end
+
+  it 'has a list of 40 of the top rated movies' do
+    VCR.use_cassette('top_movies') do
+      # top_20 = Faraday.get("https://api.themoviedb.org/3/movie/top_rated?api_key=#{ENV['MOVIES_API_KEY']}&language=en-US&page=1")
```
Comment (KieraAllen) [inline spec/features/movies/index_spec.rb:39]: I liked Austin's suggestion of possibly moving this to the top of the spec with a short comment describing why we don't need it in the spec anymore

File: spec/features/welcome_spec.rb [hunk 7]
```diff
@@ -29,6 +29,7 @@
 
       expect(page).to have_link("New to Viewing Party? Register Here")
     end
+
     it "And fail to log in is see a flash message" do
```
Comment (KieraAllen) [inline spec/features/welcome_spec.rb]: Another minor change: "log in is see" I think was meant to be "log in I see"

**Review by KieraAllen — APPROVED** [review]

File: app/controllers/movies_controller.rb (hunk 2 above)
Comment (smcunning) [reply app/controllers/movies_controller.rb]: Nice catch! Taking these out

File: app/views/movies/index.html.erb (hunk 3 above)
Comment (smcunning) [reply app/views/movies/index.html.erb]: I think you're right, adjusting that!

File: spec/features/discover_spec.rb (hunk 4 above)
Comment (smcunning) [reply spec/features/discover_spec.rb]: Added!

File: spec/features/movies/index_spec.rb (hunk 5 above)
Comment (smcunning) [reply spec/features/movies/index_spec.rb]: You got it!

File: spec/features/welcome_spec.rb (hunk 7 above)
Comment (smcunning) [reply spec/features/welcome_spec.rb]: Got it

**Review by araspaas — APPROVED** [review]
I spot no issues off initially. so i'm good to merge.

**Pull request merged**